### DISCUSSION TITLE
Migrations involving alter table

### DISCUSSION THREAD

**Discussion by kikijiki** [SeaQL/sea-query#690] [Q&A] [answered]
Assuming I have an enum like this in the first migration file:

```rust
#[derive(DeriveIden)]
pub enum Tag {
    Table,
    Id,
    Name,
}
```

I then create the table like this:

```rust
manager
    .create_table(
        Table::create()
            .table(Tag::Table)
            .if_not_exists()
            .col(
                ColumnDef::new(Tag::Id)
                    .integer()
                    .not_null()
                    .auto_increment()
                    .primary_key(),
            )
            .col(ColumnDef::new(Tag::Name).string().not_null())
            .to_owned(),
    )
    .await?;
```

Now, in a future migration I want to add a new column, `Description`.
The manager can alter the table just fine, but the existing enum will be out of date.

From the documentation I did not understand how I am supposed to handle this 🤔 
Am I supposed to edit the original declaration and add the field there?

In my mind each migration file was meant to be static, representing the DB in that point in time.
Also in case of dropping a column it would make the build fail too.

**Comment by tyt2y3** [comment, marked as answer]
It depends on where you put the enum Tag? Usually we put the enums in the same file as the migration, so that each migration file is self-contained.

**Reply by tyt2y3** [reply]
Look at this example? https://github.com/SeaQL/sea-orm/blob/master/examples/seaography_example/migration/src/m20230101_000001_create_bakery_table.rs

In which I also embedded a copy of generated entities for seeding data.

**Reply by kikijiki** [reply]
Ah I see, makes sense!

So in each new migration file I would just declare a new version of the same enum with the correct fields if needed basically.

Thank you 👍
